samples/subsys/usb/cdc_acm/src/main.c

```diff
@@ -41,12 +41,27 @@ static void interrupt_handler(struct device *dev)
 
 static void write_data(struct device *dev, const char *buf, int len)
 {
+	int sent;
+	u32_t transfer_len;
 	uart_irq_tx_enable(dev);
 
-	data_transmitted = false;
-	uart_fifo_fill(dev, (const u8_t *)buf, len);
```

Comment:
@daniel-thompson : Here's the mainline code, FYR (checking for return code).

The whole patch is also nice - it's only one more step to write byte by byte. Ain't some people smart starting with that, instead of fishing for problems from the beginning? ;-)